app/services/open_library_service.rb

```diff
@@ -0,0 +1,38 @@
+require 'httparty'
+require 'json'
+
+class OpenLibraryService
+  include Concurrent::Async
+  def initialize(type = 'ISBN', isbn = '0385472579', base_uri = 'https://openlibrary.org/api')
+    @isbn = isbn
+    @type = type
+    @base_uri = base_uri
+  end
+
+  def book
+    response = HTTParty.get("#{@base_uri}/books?bibkeys=#{@type}:#{@isbn}&format=json&jscmd=data")
+    if response.code == 200
+      hash = JSON.parse(response.body).with_indifferent_access
+      build_response(hash)
+    else
+      {
+        code: response.code,
+        message: response.message
+      }
+    end
+  end
+
+  private
+
+  def build_response(data)
+    type_isbn = "#{@type}:#{@isbn}"
```

Comment:
you can create a method called exam `type_isbn` and use this
```suggestion
   def type_isbn
     "ISBN:#{@isbn}".to_s
   end

    book_data = data[type_isbn]
```